File: src/MaksIT.Models/LetsEncryptServer/ApiKeys/ApiKeyResponse.cs
using MaksIT.Core.Abstractions.Webapi;

namespace MaksIT.Models.LetsEncryptServer.ApiKeys;

/// <summary>
/// API key metadata; <see cref="ApiKey"/> is only set when the key is created (Vault-style).
/// </summary>
public class ApiKeyResponse : ResponseModelBase {
  public Guid Id { get; set; }
  /// <summary>Plaintext key; only returned on create.</summary>
  public string ApiKey { get; set; } = string.Empty;
  public DateTime CreatedAt { get; set; }
  public string? Description { get; set; }
  public DateTime? ExpiresAt { get; set; }
  public DateTime? RevokedAt { get; set; }
}

File: src/MaksIT.WebUI/src/models/letsEncryptServer/apiKeys/ApiKeyResponse.ts
import { ResponseModelBase } from '../../ResponseModelBase'

/** Plaintext <c>apiKey</c> only on create. */
export interface ApiKeyResponse extends ResponseModelBase {
  id: string
  apiKey: string
  createdAt: string
  description?: string
  expiresAt?: string
  revokedAt?: string
}
